components/threading/global_threads/thread_menu/thread_menu.tsx

```diff
@@ -0,0 +1,145 @@
+                })}
+            >
+                {children}
+            </SimpleTooltip>
+
+            <Menu
+                ariaLabel={''}
+                openLeft={true}
+            >
+                {isFollowing ? (
+                    <Menu.ItemAction
+                        text={formatMessage({
+                            id: t('threading.threadMenu.unfollow'),
+                            defaultMessage: 'Unfollow thread',
+                        })}
+                        extraText={formatMessage({
+                            id: t('threading.threadMenu.unfollowExtra'),
+                            defaultMessage: 'You won’t be notified about replies',
+                        })}
+                        onClick={unfollow}
+                    />
+                ) : (
+                    <Menu.ItemAction
+                        onClick={follow}
+                        text={formatMessage({
+                            id: t('threading.threadMenu.follow'),
+                            defaultMessage: 'Keep following thread',
+                        })}
+                    />
+                )}
+                <Menu.ItemAction
+                    onClick={openInChannel}
+                    text={formatMessage({
+                        id: t('threading.threadMenu.openInChannel'),
+                        defaultMessage: 'Open in channel',
+                    })}
+                />
+                {hasUnreads ? (
+                    <Menu.ItemAction
+                        onClick={markRead}
+                        text={formatMessage({
+                            id: t('threading.threadMenu.markRead'),
+                            defaultMessage: 'Mark as read',
+                        })}
+                    />
+                ) : (
+                    <Menu.ItemAction
+                        onClick={markUnread}
+                        text={formatMessage({
+                            id: t('threading.threadMenu.markUnread'),
+                            defaultMessage: 'Mark as unread',
+                        })}
+                    />
+                )}
+                {isSaved ? (
+                    <Menu.ItemAction
+                        onClick={unsave}
+                        text={formatMessage({
+                            id: t('threading.threadMenu.unsave'),
+                            defaultMessage: 'Unsave',
+                        })}
+                    />
+                ) : (
+                    <Menu.ItemAction
+                        onClick={save}
+                        text={formatMessage({
+                            id: t('threading.threadMenu.save'),
+                            defaultMessage: 'Save',
+                        })}
+                    />
+                )}
+                <Menu.ItemAction
+                    onClick={copyLink}
+                    text={formatMessage({
+                        id: t('threading.threadMenu.copy'),
```

Comment:
I think with the formatMessage you don't need the `t` function there.

Reply:
It'll work without it, yes, but I think `t` is mainly for i18n tooling? (e.g. `make i18n-extract`)

Reply:
Yes, t is for translations extractions when you can't infer it from the code, in this case formatmessage is enough for the extraction tool to know that is a translatable string.

Reply:
Ahh, I didn't realize that. I thought it only picked it up `<FormattedMessage/>`.